test/common/blockchain.bats

```diff
@@ -1,5 +1,391 @@
 #!/usr/bin/env bats
 
-@test "Placeholder blockchain test script" {
-  skip
+load "${BATS_TEST_DIRNAME}/../../bats-mock/stub.bash"
+load ../test_helper
+
+setup() {
+	src_dir="${BATS_TEST_DIRNAME}/../../src"
+	testcase_dirname="$(mktemp -d)"
+
+	setup_script_dir "${src_dir}" "${testcase_dirname}"
+}
+
+teardown() {
+	cleanup_stubs
+}
+
+fill_blockchain_json() {
+	cat << EOF > blockchain.json
+{
+  "org1": {
+    "key": "key1",
+    "secret": "secret1",
+    "url": "url1",
+    "network_id": "networkid1"
+  },
+  "org2": {
+    "key": "key2",
+    "secret": "secret2",
+    "url": "url2",
+    "network_id": "networkid2"
+  }
+}
+EOF
+}
+
+cleanup_blockchain_json() {
+	rm blockchain.json
+}
+
+@test "blockchain.sh: authenticate_org should grab the specified org's information" {
+  source "${SCRIPT_DIR}/common/blockchain.sh"
+  
+	pushd "${SCRIPT_DIR}"
+
+	fill_blockchain_json
+
+	authenticate_org "org1"
+
+	[ "$BLOCKCHAIN_NETWORK_ID" = "networkid1" ]
+	[ "$BLOCKCHAIN_KEY" = "key1" ]
+	[ "$BLOCKCHAIN_SECRET" = "secret1" ]
+	[ "$BLOCKCHAIN_URL" = "url1" ]
+
+	authenticate_org "org2"
+
+	[ "$BLOCKCHAIN_NETWORK_ID" = "networkid2" ]
+	[ "$BLOCKCHAIN_KEY" = "key2" ]
+	[ "$BLOCKCHAIN_SECRET" = "secret2" ]
+	[ "$BLOCKCHAIN_URL" = "url2" ]
+
+	cleanup_blockchain_json
+	popd
+}
+
+@test "blockchain.sh: provision_blockchain should populate 'blockchain.json' without an existing service instance and key" {
+	stub cf \
+		"service bsi : cat SERVICEINFO" \
+		"create-service bsn bsp bsi : echo INSTANCECREATED" \
+		"service-key bsi bsk : cat KEYINFO" \
+		"create-service-key bsi bsk : echo KEYCREATED" \
+		"service-key bsi bsk : true"
+
+	stub tail "-n +2 : echo BLOCKCHAINJSON"
+  
+  export BLOCKCHAIN_SERVICE_INSTANCE="bsi"
+	export BLOCKCHAIN_SERVICE_NAME="bsn"
+	export BLOCKCHAIN_SERVICE_PLAN="bsp"
+	export BLOCKCHAIN_SERVICE_KEY="bsk"
+
+  source "${SCRIPT_DIR}/common/blockchain.sh"
+
+	run provision_blockchain
+	[ $status -eq 0 ]
+  [ "${lines[0]}" = "INSTANCECREATED" ]
```

Comment:
if you add set -x to functions this will change,  Instead of testing for a specific line I would test to see if the string in the overall output contains INSTANCECREATED